apps/tailwind-components/components/form/FieldInput.vue

```diff
@@ -0,0 +1,50 @@
+<script setup lang="ts">
+import type { InputString } from "#build/components";
+import type {
+  columnId,
+  columnValue,
+  ValueType,
+} from "../../../metadata-utils/src/types";
+
+defineProps<{
+  type: ValueType;
+  id: columnId;
+  label: string;
+  data: columnValue;
+}>();
+
+defineEmits(["focus", "input", "error", "update:modelValue"]);
+defineExpose({ validate });
+
+const input = ref<InstanceType<typeof InputString>>();
+
+function validate(value: columnValue) {
+  if (input && input.value && input.value.validate) {
+    input.value.validate(value as string);
```

Comment:
why the cast, wouldn't it be better to actually change to a string? Like using JSON.stringify?

Reply:
I know, but casting it like this is dangerous since it will hide type mismatches. 

On another note, (not the PR for it, I know) but we should have a naming convention for types as we do for interfaces I think. TColumnValue or columnValueType or...